src/linkpresentation.cs

```diff
@@ -0,0 +1,99 @@
+//
+// LinkPresentation C# bindings
+//
+// Authors:
+//	TJ Lambert  <t-anlamb@microsoft.com>
+//
+// Copyright 2019 Microsoft Corporation All rights reserved.
+//
+
+#if MONOMAC
+using AppKit;
+using UIView = AppKit.NSView;
+#else
+using UIKit;
+#endif
+
+using System;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+
+namespace LinkPresentation {
+
+	[ErrorDomain ("LPErrorDomain")]
+	[NoWatch]
+	[Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[Native]
+	public enum LPErrorCode : long {
+		Unknown = 1,
+		MetadataFetchFailed,
+		MetadataFetchCancelled,
+		MetadataFetchTimedOut,
+	}
+
+	[NoWatch]
+	[Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	interface LPLinkMetadata : NSCopying, NSSecureCoding {
+
+		[NullAllowed, Export ("originalURL", ArgumentSemantic.Retain)]
+		NSUrl OriginalUrl { get; set; }
+
+		[NullAllowed, Export ("URL", ArgumentSemantic.Retain)]
+		NSUrl Url { get; set; }
+
+		[NullAllowed, Export ("title", ArgumentSemantic.Copy)]
+		string Title { get; set; }
+
+		[NullAllowed, Export ("iconProvider", ArgumentSemantic.Retain)]
+		NSItemProvider IconProvider { get; set; }
+
+		[NullAllowed, Export ("imageProvider", ArgumentSemantic.Retain)]
+		NSItemProvider ImageProvider { get; set; }
+
+		[NullAllowed, Export ("videoProvider", ArgumentSemantic.Retain)]
+		NSItemProvider VideoProvider { get; set; }
+
+		[NullAllowed, Export ("remoteVideoURL", ArgumentSemantic.Retain)]
+		NSUrl RemoteVideoUrl { get; set; }
+	}
+
+	[NoWatch]
+	[Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (UIView))]
+	interface LPLinkView {
+
+		[Export ("initWithFrame:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (CGRect rect);
```

Comment:
good call (in general) but it's not clear if needed
Does iOS let you control the view UI size ? (it does not always for system views)

Reply:
So if I call initWithFrame: CGRectMake (5,5,5,5)
I can create a LPLinkView with the size 5,5 at dimensions 5,5

Reply:
sounds legit too :)